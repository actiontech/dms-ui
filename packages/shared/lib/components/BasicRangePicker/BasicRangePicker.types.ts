import { RangePickerProps } from 'antd/es/date-picker';
import { ReactNode } from 'react';

export type BasicRangePickerProps = RangePickerProps & {
  prefix?: ReactNode;
  hideSuperIcon?: boolean;
};
