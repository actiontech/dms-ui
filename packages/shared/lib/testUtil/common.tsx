/* eslint-disable no-console */
export const ignoreAntdFormValidateWarning = () => {
  const warning = console.warn;

  beforeAll(() => {
    console.warn = (...arg) => {
      if (typeof arg[0] === 'string' && arg[0].includes('async-validator:')) {
        return;
      }
      warning(...arg);
    };
  });

  afterAll(() => {
    console.warn = warning;
  });
};

export const ignoreAntdUseFormNotConnectedError = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'Instance created by `useForm` is not connected to any Form element. Forget to pass `form` prop?'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

export const ignoreComponentUncontrolled = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'A component is changing a controlled input to be uncontrolled.'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

// 当为原生的 html 添加大驼峰、小驼峰之类的自定义属性，会出现报错
export const ignoreComponentCustomAttr = () => {
  const error = console.error;
  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

/**
 * @description: 在选择 date-range-pick 组件的结束日期 input 元素时，实际选择到了，报此 warning
 */
export const ignoreComponentWarnExpDatePickChooseElement = () => {
  const warn = console.warn;
  beforeAll(() => {
    console.warn = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'Not match any format. Please help to fire a issue about this.'
        )
      ) {
        return;
      }
      warn(...arg);
    };
  });

  afterAll(() => {
    console.warn = warn;
  });
};

export const ignoreComponentAutoCreatedListNoKey = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes('Each child in a list should have a unique "key" prop')
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};
