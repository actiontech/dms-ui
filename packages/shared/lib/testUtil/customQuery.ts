export {
  getBySelector,
  queryBySelector,
  getAllBySelector,
  getSelectContentByFormLabel,
  getSelectValueByFormLabel,
  selectOptionByIndex,
  selectCustomOptionByClassName,
  getValueByInput,
  sleep,
  getHrefByText,
  getAllHrefByText
} from '@actiontech/dms-kit/src/testUtil/customQuery';
