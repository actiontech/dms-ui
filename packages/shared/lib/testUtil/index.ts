export * from './superRender';
export * from './mockHook';
export * from './mockApi';
export * from './customQuery';
export * from './mockRedux';
export * from './common';
export * from './LocationDisplay';
