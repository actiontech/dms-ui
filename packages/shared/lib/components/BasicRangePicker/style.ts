import { styled } from '@mui/material/styles';
import { DatePicker } from 'antd5';
import { ANTD_PREFIX_STR } from '../../data/common';

export const BasicRangePickerStyleWrapper = styled(DatePicker.RangePicker)`
  &.${ANTD_PREFIX_STR}-picker-range.basic-range-picker-wrapper {
    .${ANTD_PREFIX_STR}-picker-clear {
      background-color: transparent;
      opacity: 1;

      .custom-icon-close {
        color: ${({ theme }) => theme.sharedTheme.basic.colorDefaultIcon};
      }
    }

    .${ANTD_PREFIX_STR}-picker-range-separator {
      .custom-icon-arrow-right {
        color: ${({ theme }) => theme.sharedTheme.basic.colorDefaultIcon};
      }
    }
  }

  &.custom-picker-prefix.basic-range-picker-wrapper {
    .${ANTD_PREFIX_STR}-picker-suffix {
      margin-left: 1px;
      margin-right: 10px;
      order: -1;
    }
  }

  &.${ANTD_PREFIX_STR}-picker-range:not(.${ANTD_PREFIX_STR}-picker-borderless) {
    border: ${({ theme }) =>
      theme.sharedTheme.components.basicRangePicker.default.border};
    border-radius: 4px;
  }

  .${ANTD_PREFIX_STR}-picker-input {
    input::placeholder {
      color: ${({ theme }) =>
        theme.sharedTheme.components.basicRangePicker.default.placeholder
          .color};
    }
  }

  &.${ANTD_PREFIX_STR}-picker-range:hover:not(.${ANTD_PREFIX_STR}-picker-focused):not(
      .${ANTD_PREFIX_STR}-picker-borderless
    ):not(.${ANTD_PREFIX_STR}-picker-status-error):not(
      .${ANTD_PREFIX_STR}-picker-disabled
    ) {
    border: ${({ theme }) =>
      theme.sharedTheme.components.basicRangePicker.hover.border};
  }

  &.${ANTD_PREFIX_STR}-picker-focused:not(.${ANTD_PREFIX_STR}-picker-borderless) {
    border: ${({ theme }) =>
      theme.sharedTheme.components.basicRangePicker.active.border};
  }

  &.${ANTD_PREFIX_STR}-picker-status-error:not(.${ANTD_PREFIX_STR}-picker-borderless) {
    border: ${({ theme }) =>
      theme.sharedTheme.components.basicRangePicker.error.border};
  }

  &.${ANTD_PREFIX_STR}-picker-disabled:not(.${ANTD_PREFIX_STR}-picker-borderless) {
    border: ${({ theme }) =>
      theme.sharedTheme.components.basicRangePicker.disabled.border};
  }
`;
