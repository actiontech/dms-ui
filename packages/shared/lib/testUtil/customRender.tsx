import { render, renderHook } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps
} from 'react-router-dom';
import { ConfigProvider, theme as antdTheme } from 'antd';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary, IStore } from '../types/common.type';
import { storeFactory } from './mockRedux';
import { StyledEngineProvider, ThemeProvider } from '@mui/system';
import { mount } from 'enzyme';

import lightTheme from '../theme/light';

type MountParams = Parameters<typeof mount>;
type RenderParams = Parameters<typeof render>;
type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={lightTheme}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: IStore
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <Provider store={storeFactory(storeState)}>{children}</Provider>
    )
  });
};

export const renderHooksWithReduxAndRouter = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: IStore
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <Provider store={storeFactory(storeState)}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    )
  });
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <RecoilRoot {...otherProps?.recoilRootProps}>
          <Provider store={storeFactory(otherProps?.initStore)}>
            <MemoryRouter {...otherProps?.routerProps}>
              <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
            </MemoryRouter>
          </Provider>
        </RecoilRoot>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme
    }
  });
};
