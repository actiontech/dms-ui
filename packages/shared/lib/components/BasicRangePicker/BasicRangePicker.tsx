import { ConfigProvider } from 'antd';
import classnames from 'classnames';
import { BasicRangePickerStyleWrapper } from './style';
import { ComponentControlHeight } from '../../data/common';
import { BasicDatePickerDropDownStyleWrapper } from '../BasicDatePicker/style';
import { CloseOutlined, RightOutlined, LeftOutlined } from '@actiontech/icons';
import { BasicRangePickerProps } from './BasicRangePicker.types';

const BasicRangePicker: React.FC<BasicRangePickerProps> = (props) => {
  const {
    className,
    suffixIcon,
    prefix,
    hideSuperIcon = true,
    ...otherParams
  } = props;

  return (
    <ConfigProvider
      theme={{
        components: {
          DatePicker: {
            controlHeight: ComponentControlHeight.default,
            controlHeightLG: ComponentControlHeight.lg,
            controlHeightSM: ComponentControlHeight.sm
          }
        }
      }}
    >
      <BasicDatePickerDropDownStyleWrapper hideSuperIcon={hideSuperIcon}>
        <BasicRangePickerStyleWrapper
          className={classnames('basic-range-picker-wrapper', className, {
            'custom-picker-prefix': !suffixIcon && !!prefix
          })}
          {...otherParams}
          /**
           * when both suffixIcon and prefix exist, the former will force the latter to be overwritten
           */
          suffixIcon={suffixIcon ?? prefix}
          //todo allowClear={{clearIcon: <IconClose />}} need antd 5.8.0
          clearIcon={<CloseOutlined fill="currentColor" />}
          separator={
            <RightOutlined width={14} height={14} fill="currentColor" />
          }
          getPopupContainer={(trigger) => {
            return trigger;
          }}
          nextIcon={
            <span className="next-icon">
              <RightOutlined width={14} height={14} />
            </span>
          }
          prevIcon={
            <span className="prev-icon">
              <LeftOutlined width={14} height={14} />
            </span>
          }
        />
      </BasicDatePickerDropDownStyleWrapper>
    </ConfigProvider>
  );
};

export default BasicRangePicker;
