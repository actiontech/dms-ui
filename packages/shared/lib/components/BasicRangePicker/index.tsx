import { ConfigProvider } from 'antd';
import { IconArrowRight, IconClose } from '../../Icon';
import { RangePickerProps } from 'antd/es/date-picker';
import classnames from 'classnames';
import { ReactNode } from 'react';
import { BasicRangePickerStyleWrapper } from './style';
import { ComponentControlHeight } from '../../data/common';
import { IconDatePickerLeft, IconDatePickerRight } from '../../Icon/common';
import { BasicDatePickerDropDownStyleWrapper } from '../BasicDatePicker/style';

export type BasicRangePickerProps = RangePickerProps & {
  prefix?: ReactNode;
  hideSuperIcon?: boolean;
};

const BasicRangePicker = (props: BasicRangePickerProps) => {
  const {
    className,
    suffixIcon,
    prefix,
    hideSuperIcon = true,
    ...otherParams
  } = props;

  return (
    <ConfigProvider
      theme={{
        components: {
          DatePicker: {
            controlHeight: ComponentControlHeight.default,
            controlHeightLG: ComponentControlHeight.lg,
            controlHeightSM: ComponentControlHeight.sm
          }
        }
      }}
    >
      <BasicDatePickerDropDownStyleWrapper hideSuperIcon={hideSuperIcon}>
        <BasicRangePickerStyleWrapper
          className={classnames('basic-range-picker-wrapper', className, {
            'custom-picker-prefix': !suffixIcon && !!prefix
          })}
          {...otherParams}
          /**
           * when both suffixIcon and prefix exist, the former will force the latter to be overwritten
           */
          suffixIcon={suffixIcon ?? prefix}
          //todo allowClear={{clearIcon: <IconClose />}} need antd 5.8.0
          clearIcon={<IconClose />}
          separator={<IconArrowRight />}
          getPopupContainer={(trigger) => {
            return trigger;
          }}
          nextIcon={<IconDatePickerRight className="next-icon" />}
          prevIcon={<IconDatePickerLeft className="prev-icon" />}
        />
      </BasicDatePickerDropDownStyleWrapper>
    </ConfigProvider>
  );
};

export default BasicRangePicker;
