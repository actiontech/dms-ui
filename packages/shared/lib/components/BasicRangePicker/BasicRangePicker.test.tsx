import { renderWithTheme } from '../../testUtil/customRender';
import BasicRangePicker from './BasicRangePicker';
import { BasicRangePickerProps } from './BasicRangePicker.types';

describe('lib/BasicRangePicker', () => {
  const customRender = (params: BasicRangePickerProps) => {
    return renderWithTheme(<BasicRangePicker {...params} />);
  };

  it('render diff size range date', () => {
    const { baseElement } = customRender({
      className: 'custom-date-range',
      size: 'small'
    });
    expect(baseElement).toMatchSnapshot();

    const { baseElement: baseElement1 } = customRender({
      size: 'middle'
    });
    expect(baseElement1).toMatchSnapshot();

    const { baseElement: baseElement2 } = customRender({
      size: 'large'
    });
    expect(baseElement2).toMatchSnapshot();
  });

  it('render none prefix icon', () => {
    const { baseElement } = customRender({
      className: 'custom-date-range',
      suffixIcon: false,
      prefix: false
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('render no supper icon', () => {
    const { baseElement } = customRender({
      className: 'custom-date-range',
      hideSuperIcon: false
    });
    expect(baseElement).toMatchSnapshot();
  });
});
