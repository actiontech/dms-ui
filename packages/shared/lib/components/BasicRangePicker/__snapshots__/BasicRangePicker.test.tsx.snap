// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/BasicRangePicker should match snapshot 1`] = `
<div>
  <div
    class="css-1y8q0oh"
  >
    <div
      class="ant-picker ant-picker-range basic-range-picker-wrapper css-w7y1zu"
    >
      <div
        class="ant-picker-input ant-picker-input-active"
      >
        <input
          autocomplete="off"
          placeholder="开始日期"
          readonly=""
          size="12"
          value=""
        />
      </div>
      <div
        class="ant-picker-range-separator"
      >
        <svg
          fill="currentColor"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.684 7 4.796 4.112l.825-.824L9.334 7 5.62 10.712l-.825-.825z"
          />
        </svg>
      </div>
      <div
        class="ant-picker-input"
      >
        <input
          autocomplete="off"
          placeholder="结束日期"
          readonly=""
          size="12"
          value=""
        />
      </div>
      <div
        class="ant-picker-active-bar"
        style="left: 0px; width: 0px; position: absolute;"
      />
    </div>
  </div>
</div>
`;
